projects/make-a-vis/src/lib/data-view/shared/export-table.service.ts

```diff
@@ -1,28 +1,45 @@
-import { Injectable } from '@angular/core';
+import { Injectable, OnDestroy } from '@angular/core';
 import { saveAs } from 'file-saver';
 import { at as loAt, map as loMap } from 'lodash';
 import { unparse } from 'papaparse';
 
 import { DataSource } from './data.service';
+import { Subscription } from 'rxjs';
 
 /**
  * Service for saving a table's data to a csv file.
  */
 @Injectable({
   providedIn: 'root'
 })
-export class ExportTableService {
+export class ExportTableService implements OnDestroy {
+  /**
+   * Data subscription of export table service
+   */
+  private dataSubscription: Subscription;
+
   /**
    * Saves table data to a csv file.
    *
    * @param source The table data.
    */
-  save(source: DataSource): void {
-    const header = loMap(source.columns, 'label');
-    const ids = loMap(source.columns, 'id');
-    const values = loMap(source.data, item => loAt(item, ids));
-    const csv = unparse({ fields: header, data: values });
-    const blob = new Blob([csv], { type: 'text/csv' });
-    saveAs(blob, source.label + '.csv');
+   save(source: DataSource): void {
+    this.dataSubscription = source.data.subscribe((data) => {
```

Comment:
subscription leak. Check if a dataSubscription is already there and unsubscribe. Or better yet, have the subscriber close it after one emission (takeOne) that way subscriptions unsubscribe automatically.